### ISSUE TITLE
Launch Protractor multiple times from same nodejs execution

### ISSUE THREAD

**Issue by carrascoMDD** [angular/protractor#2836]
To integrate Protractor AngularJS tests, in an end2end test tool  (https://www.npmjs.com/package/asyncshell  https://github.com/carrascoMDD/asyncshell ), which interacts with JSON APIs and AngularUIs, Protractor is/may be launched multiple times, from the same operating system launch of nodejs.
While I came up with a smallish upgrade/hack of 3/4 changes into Protractor code base, which allows to reset state in Protractor before runs after the first one, such that Protractor runs as "freshly" launched, this was on to Protractor 1.3.1 .
Indeed, I want to integrate recent releases of Protractor before releasing the "asyncshell" end2end test framework to the Community.
I wonder if more recent releases of Protractor already incorporate a resetting feature allowing repeated running of Protractor in the same operating system launch of NodeJS, such that I do not distract the Community at large with pull request of possibly similar changes to the latest version of Protractor.

**Comment by NickTomlin** [comment]
At the moment I do not know of a way to do this without changing the protractor internals. You could potentially use [`runner`](https://github.com/angular/protractor/blob/master/lib/runner.js) directly but some state may still be preserved :\

**Comment by juliemr** [comment]
Why can't you just run each launch in as a separate child process? That will guarantee that it's completely reset.

**Comment by carrascoMDD** [comment]
The case for AsyncShell is to run really end to end scenarios, testing both APIs (i.e. REST JSON) and UIs (i.e. AngularJS). AsyncShell supports a nested hierarchy of contextual property environments, both read and written by individual tests, such that values, whole objects and references can be passed along different tests. 
While some off-process machinery could be put in place, to pass values, objects and references among processes, rich object structures would be hard to handle. Therefore, a single process seems better suited for the integration purpose (even if memory consumption required the implementation of some  memory management instrumentation).
Please note that I am still working on productizing AsyncShell for reuse by the willing community at large. In the meanwhile, I've uploaded a 20 minutes video, illustrating the intent.   
https://youtu.be/mxIb9Imy5tI

**Comment by juliemr** [comment]
I think your best bet then would to be to use your own [runner](https://github.com/angular/protractor/blob/master/lib/runner.js), as @NickTomlin mentioned. 

I'm going to go ahead and close this issue as an answered question, since there's no specific actionable items for us. If you have specific feature requests, please open up new issues for them. Thanks!

**Comment by carrascoMDD** [comment]
Thanks. I understand then, that the answer is:
Protractor does not support the Feature of running multiple times within the same nodejs process.
I fear I did not describe the solution at all - my apologies. I shall endeavour to explain myself better next time. Indeed, the actionable item may affect more than just the runned, but also jasmine and minijasmine.
I shall open a new issue, and possibly a pull request - if I find the way to do it from the 1.3.1 version, or can manage to re-do it over the latest protractor version.

The solution I hacked can be observed by searching for the string
ACV OJO
in files
https://github.com/carrascoMDD/asyncshell/blob/master/lib/protractor-131-relaunchable/lib/runner.js
https://github.com/carrascoMDD/asyncshell/blob/master/lib/protractor-131-relaunchable/lib/frameworks/jasmine.js
https://github.com/carrascoMDD/asyncshell/blob/master/lib/protractor-131-relaunchable/node_modules/minijasminenode/lib/index.js
And is used from 
https://github.com/carrascoMDD/asyncshell/blob/master/lib/browser-test/protractor-test.js
from the method
pS_ProtractorTest_Retrying
with sentences
                var aRunnerResetter = this.fRunnerResetterFunction( aProtractorRunner);
and
aProtractorRunner
                    .run( aRunnerResetter ) /* ACV OJO 201410262213 to reset info held from previous protractor Runner and jasmine executions in same node js process.*/
                    .then( function( theExitCode) { ...
